x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/BinaryPlan.java

```diff
@@ -45,11 +43,6 @@ public final BinaryPlan replaceRight(LogicalPlan newRight) {
         return replaceChildren(left, newRight);
     }
 
-    protected AttributeSet computeReferences() {
-        // TODO: this needs to be driven by the join config
-        return Expressions.references(output());
```

Comment:
This made no sense. And calling `output()` here is bad because `references()` is called in the `fieldNames` method - at which point the `output()` method is illegal to use because it requires a resolved plan.